cognite/client/_api/files.py

```diff
@@ -153,9 +153,8 @@ def create(self, file_metadata: FileMetadata, overwrite: bool = False) -> Tuple[
             url_path=self._RESOURCE_PATH, json=file_metadata.dump(camel_case=True), params={"overwrite": overwrite}
         )
         returned_file_metadata = res.json()
-        upload_url = returned_file_metadata.pop("uploadUrl")
+        upload_url = returned_file_metadata["uploadUrl"]
```

Comment:
Tests for these methods started failing due to caching .json() response. uploadUrl was removed from the cached dict, so couldn't use it to compare with expected object. No need for pop here since when loading this dict into FileMetadata object this field is ignored.